Source/Fuse.Reactive.JavaScript/JavaScript.Dependencies.uno

```diff
@@ -0,0 +1,155 @@
+using Uno;
+using Uno.UX;
+using Uno.Collections;
+using Uno.Compiler;
+using Fuse.Scripting;
+using Uno.Testing;
+using Uno.Threading;
+
+namespace Fuse.Reactive
+{
+	public partial class JavaScript
+	{
+		public class Dependency : IListener
+		{
+			internal string Name { get; private set; }
+			internal IExpression Expression { get; private set; }
+
+			[UXConstructor]
+			public Dependency([UXParameter("Name")] string name, [UXParameter("Expression")] IExpression expression)
+			{
+				Name = name;
+				Expression = expression;
+			}
+
+			JavaScript _script;
+			IDisposable _expSubscription;
+
+			internal void Subscribe(JavaScript script)
+			{
+				Unsubscribe();
+				_script = script;
+				_expSubscription = Expression.Subscribe(script, this);
+			}
+
+			internal void Unsubscribe()
+			{
+				if (_expSubscription != null)
+				{
+					_expSubscription.Dispose();
+					_expSubscription = null;
+				}
+				Value = null;
+				HasValue = false;
+				_script = null;
+			}
+
+			internal bool HasValue;
+			internal object Value;
+
+			void IListener.OnNewData(IExpression source, object data)
+			{
+				if (_script == null) return;
+
+				Value = data;
+				HasValue = true;
+
+				_script.DispatchEvaluateIfDependenciesReady();
+			}
+		}
+
+
+   		List<Dependency> _dependencies;
+		/** A list of named expressions that will be evaluated and injected as variables into the script.
+		
+			This property allows injecting dependencies defined as UX expressions into the script using the `dep:` XML namespace.
+			
+			Example:
+
+				<JavaScript>
+					exports.foo = 123
+				</JavaScript>
+				<JavaScript dep:foo="{foo}">
+					foo // this is now 123
+				</JavaScript>
+
+			A script is not executed until all of its dependencies are available. If any of the dependencies change, the script is re-executed.
+
+			This has multiple use-cases:
+			* Accessing data from data context `dep:foo="{foo}"`
+			* Accessing properties synchronously `dep:SomeProp="{Property SomeProp}"`
+		*/
+		public IList<Dependency> Dependencies 
+		{ 
+			get 
+			{
+				if (_dependencies == null) 
+					_dependencies = new List<Dependency>();
+
+				return _dependencies; 
+			} 
+		}
+
+		void SubscribeToDependenciesAndDispatchEvaluate() 
+		{
+			if (_dependencies != null) 
+				for (var i = 0; i < _dependencies.Count; i++)
+					_dependencies[i].Subscribe(this);
+
+			DispatchEvaluateIfDependenciesReady();
+		}
+
+		void DisposeDependencySubscriptions()
+		{
+			if (_dependencies != null) 
+				for (var i = 0; i < _dependencies.Count; i++)
+					_dependencies[i].Unsubscribe();
+		}
+
+		void DispatchEvaluateIfDependenciesReady()
```

Comment:
I think you need to defer this otherwise you'll end up evaluating the module several times needlessly.

Consider a `<JavaScript deps:a="{a}" deps:b="{b}">` If both `a` and `b` change you'll end up instantiating the module twice, rather than just once. Since the values to a module are often related I think it's very likely that multiple dependencies will change at one time.